src/test/framework/test_env_builder_kubernetes.js

```diff
@@ -222,7 +224,8 @@ async function run_single_test_env(params) {
         console.log(`test ${test_name} failed. ${err}`);
     }
 
-    if (clean || clean_single_test) {
+    //Will not delete the namespace if the test has failed.
+    if (!test_failed && (clean || clean_single_test)) {
```

Comment:
I think there should be a flag to control whether we keep namespaces on failure. 
on weak clusters (e.g. minikube) if we keep even one failing namespace we can cause all other tests to fail on resources